staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -187,6 +181,41 @@ func ValidateCustomResourceDefinitionSpec(spec *apiextensions.CustomResourceDefi
 		}
 	}
 
+	ValidateCustomResourceConversion(spec.Conversion, fldPath.Child("conversion"))
+
+	return allErrs
+}
+
+func validateEnumStrings(fldPath *field.Path, value string, accepted []string, required bool) field.ErrorList {
+	if required && value == "" {
```

Comment:
nit: this logic means a caller would have to include "" in `accepted` to validate an optional enum, which seems slightly odd (both current callers set required==true, so the oddity isn't apparent in this PR)